Label-Microservice/notebooks/Training_Pipeline.ipynb

```diff
@@ -0,0 +1,587 @@
+      "Layer sha256:6669e38ab1bac2750ee43e9fd769478543e5ee6866cfbbe10c5f55584681792d exists, skipping\n",
+      "Layer sha256:13806a5c262364d945866dc42e584c11259483e90e367e20d7f9ff2bc9401aed exists, skipping\n",
+      "Layer sha256:5b6ac7f35d3dce1cd93b9740c7342fe317b179c5ebb8d9e92069d5c671a756ae exists, skipping\n",
+      "Layer sha256:6abc03819f3e00a67ed5adc1132cfec041d5f7ec3c29d5416ba0433877547b6f exists, skipping\n",
+      "Layer sha256:46e6314d080196417906065be01b63c7dc3981caddd1b01b6395bbed5b4147d8 pushed.\n",
+      "Layer sha256:b92dd3e4292d4390c5bf0795a5424fc1a8ae7d44d143153157d27ad325530168 pushed.\n",
+      "Finished upload of: gcr.io/issue-label-bot-dev/training/fairing-job:5EDD484A\n",
+      "Pushed image gcr.io/issue-label-bot-dev/training/fairing-job:5EDD484A in 4.249339093978051s.\n"
+     ]
+    }
+   ],
+   "source": [
+    "builder = append.append.AppendBuilder(registry=DOCKER_REGISTRY,\n",
+    "                                      base_image=cluster_builder.image_tag,\n",
+    "                                      preprocessor=preprocessor)\n",
+    "builder.build()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Build pipeline\n",
+    "Kubeflow [Pipelines](https://www.kubeflow.org/docs/pipelines/) builds reusable end-to-end machine learning workflows.\n",
+    "\n",
+    "Define the pipeline as a Python function. \"@kfp.dsl.pipeline\" is a required decoration including name and description properties.\n",
+    "\n",
+    "We define two steps for our training pipelines, including scrapping issues and training model, both of which will be executed in our built image from Kubeflow Fairing. Also, we use GPU and add GCP credentials to the pipelines."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import kfp\n",
+    "import kfp.components as comp\n",
+    "import kfp.gcp as gcp\n",
+    "import kfp.dsl as dsl\n",
+    "import kfp.compiler as compiler"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "target_image = 'gcr.io/issue-label-bot-dev/training/fairing-job:5EDD484A'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@dsl.pipeline(\n",
+    "   name='Training pipeline',\n",
+    "   description='A pipeline that loads embeddings and trains a model for a github repo.'\n",
+    ")\n",
+    "def train_pipeline(owner, repo):\n",
+    "    scrape_op = dsl.ContainerOp(\n",
+    "            name='scrape issues',\n",
+    "            image=target_image,\n",
+    "            command=['python', 'Issues_Loader.py', 'save_issue_embeddings', f'--owner={owner}', f'--repo={repo}'],\n",
```

Comment:
Nit: `Issues_Loader` is not good naming style. Either `issues_loader` in snake case or `IssuesLoader` in camel case.

Reply:
Will fix.